extensions/ql-vscode/src/remote-queries/gh-api/gh-api-client.ts

```diff
@@ -0,0 +1,86 @@
+import { Credentials } from '../../authentication';
+import { OctokitResponse } from '@octokit/types/dist-types';
+import { VariantAnalysisSubmission } from '../shared/variant-analysis';
+import {
+  VariantAnalysis,
+  VariantAnalysisQueryLanguage,
+  VariantAnalysisRepoTask,
+  VariantAnalysisSubmissionRequest
+} from './variant-analysis';
+
+export async function submitVariantAnalysis(
+  credentials: Credentials,
+  submissionDetails: VariantAnalysisSubmission
+): Promise<VariantAnalysis> {
+  const octokit = await credentials.getOctokit();
+
+  const { actionRepoRef, query, databases, controllerRepoId } = submissionDetails;
+
+  const data: VariantAnalysisSubmissionRequest = {
+    action_repo_ref: actionRepoRef,
+    language: query.language as VariantAnalysisQueryLanguage,
```

Comment:
I don't think the casting is necessary if the enum and the string union match in the values they accept. By skipping the casting here, we will catch errors when they are out-of-sync.

```suggestion
    language: query.language,
```